processor/k8sattributesprocessor/config.go

```diff
@@ -7,7 +7,7 @@ import (
 	"fmt"
 	"regexp"
 
-	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	conventions "go.opentelemetry.io/collector/semconv/v1.21.0"
```

Comment:
I don't feel comfortable bumping the conventions as part of this PR, that should be its own change and should be highly scrutinized for breaking changes.  For now, instead, can you add the new conventions as constants in this component? I think we do that for some others already.

Reply:
Fair enough - will do.

Reply:
Note: there's already some inconsistency in the versions in k8sattributesprocessor.  Most imports are at 1.6.1, but `processor.go` and `processor_test.go` are at 1.8.0.